### DISCUSSION TITLE
Accessing other Netbox data within Jinja2 configuration generation templates

### DISCUSSION THREAD

**Discussion by aliclubb** [netbox-community/netbox#12783] [Help Wanted!]
Hi everyone! Is there a way to access Netbox data outside of `device.blah` from within Jinja2 configuration templates? i.e. do things like
```
{% set another_device = Device.objects.get(id=19) %}
{% set a_prefix = Prefix.objects.get(id=7) %}
{{ another_device.name }}
{{ a_prefix.prefix }}
```

I would basically like to be able to do read-only Django commands from within Jinja2 configuration templates. (I think that's the best way to linguistically describe what I want, but I'm not a pro-Pythoneer!)

**Comment by boxops** [comment]
So far I have only been able to access other devices if a local device object (where the template is rendered from) is associated to another device somehow, for example it has a connection. I wrote the following Jinja2 Config Template and assigned 2 devices to render:

hostname {{ device.name }}
{%- for interface in device.interfaces.all() %}
Cable: {{ interface.cable }}
Cable end: {{ interface.cable_end }}
{%- for endpoint in interface.connected_endpoints %}
Connected endpoint: {{ endpoint }}
Connected device: {{ endpoint.device }}
{%- endfor %}
{%- endfor %}

Render Config output from "test device":

Hostname: test device
Cable: #1
Cable end: B
Connected endpoint: 1
Connected device: test device 2

Render Config  output from "test device 2":

Hostname: test device 2
Cable: #1
Cable end: A
Connected endpoint: 1
Connected device: test device


PS: I also would like to know how to access objects/devices from other objects that are not associated to any other objects.
